### ISSUE TITLE
TreeMap: legend.valueLabels does't work

### ISSUE THREAD

**Issue by Identity-labs** [amcharts/amcharts4#1084]
hi,

On treemap this configuration does't work
We can't display value in legend
This error appear : 

> Cannot read property 'value' of undefined

`{
	legend: {
		markers: {
			width: 9,
			height: 9,
			cornerRadius: 9,
			strokeWidth: 0
		},
		labels: {
			position: "right",
			template: {
				text: "{name}"
			}
		},
		valueLabels: {
			template: {
				text: "{value}"
			}
		},
		fontSize: "10"
	}
}`

Thx

**Comment by martynasma** [comment]
Hm, looks like a bug. We'll see if this can be fixed.

Working example for dev team:
https://codepen.io/team/amcharts/pen/be2d6fefc7bab4214690ccfc76b4c5fa?editors=0010

**Comment by Identity-labs** [comment]
When we set useDefaultMarker a true, we don't have anymore that error, but still valueLabels don't show.

**Comment by Identity-labs** [comment]
Do you have a deadline to solve this bug ?

**Comment by Identity-labs** [comment]
Here the trace
```
TypeError: Cannot read property 'value' of undefined
    at e.get [as value] (charts.js:20)
    at e.getTagValueFromObject (core.js:19)
    at e.getTagValue (core.js:19)
    at e.populateString (core.js:19)
    at e.updateCurrentText (core.js:19)
    at e.draw (core.js:19)
    at e.validate (core.js:19)
    at core.js:19
    at Object.e.each (core.js:19)
    at t.each (core.js:19)
```

**Comment by martynasma** [comment]
Fixed in [4.3.6](https://github.com/amcharts/amcharts4/releases/tag/4.3.6).

## [4.3.6] - 2019-04-04

### Added
- New `DateFormatter` property: `timezoneOffset`. Set it to a number of minutes (to UTC) to recalculate all date/time to a specific time zone.

### Changed
- `DateFormatter` will now use locale information for its default date format set in `dateFormat`. Format in default "International English" locale did not change, and remains at `"yyyy-MM-dd"`. Other locales might or might not produce different date formats.
- Improved `ValueAxis` scale calculation algorithm when `min`/`max` is set.
- `"hit"` event is now added on `FlowDiagramNode` template instead of the class itself. This allows disabling default toggle/drag behavior and replacing with own events.

### Fixed
- Themes were not properly being applied to bullets.
- `Popup.draggable` was being ignored.
- `Series.legendSettings.itemValueText` was being ignored when `cursorTooltipEnabled = false`. #1141
- Mouse events over `LineSeries` segments where happening at wrong position.
- `legend.valueLabels` on `TreeMap` did not work.
- Changing Map projection with `panBehavior = "rotateLongLat"` could cause a JS error.
- Changing `TreeMap` data after chart size changed could cause zoom-out button to appear.
- Solved flickering of stacked columns when raw data was invalidated.
- Updating data of a `TreeMap` when animation was playing could result in a JS error.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
